src/app/core/directives/scroll.directive.ts

```diff
@@ -0,0 +1,67 @@
+import { Directive, ElementRef, Input, OnChanges } from '@angular/core';
+
+/**
+ * Structural directive.
+ * Used on an element, the elements parent will scroll to it, when the given value is true
+ *
+ * @example
+ * <div [ishScroll]="true" [scrollDuration]="500">
+ *   Parent will scroll smoothly to this element within 500 milliseconds
+ * </div>
+ * or
+ * <div [ishScroll]="true">
+ *   Parent will scroll to this element instantly
+ * </div>
+ * or
+ * <div [ishScroll]="false">
+ *   Parent will not scroll to this element
+ * </div>
+ */
+@Directive({
+  selector: '[ishScroll]',
+})
+export class ScrollDirective implements OnChanges {
```

Comment:
This directive is quite math-heavy. I think it's important that it is well understood what goes on here so please add some unit tests in addition to more documentation.